libnetwork/drivers/bridge/bridge_linux.go

```diff
@@ -553,8 +553,21 @@ func (d *driver) configure(option map[string]interface{}) error {
 
 var newFirewaller = func(ctx context.Context, config firewaller.Config) (firewaller.Firewaller, error) {
 	if nftables.Enabled() {
-		return nftabler.NewNftabler(ctx, config)
+		fw, err := nftabler.NewNftabler(ctx, config)
+		if err != nil {
+			return nil, err
+		}
+		// Without seeing config (interface names, addresses, and so on), the iptabler's
+		// cleaner can't clean up network or port-specific rules that may have been added
+		// to iptables built-in chains. So, if cleanup is needed, give the cleaner to
+		// the nftabler. Then, it'll use it to delete old rules as networks are restored.
+		fw.(firewaller.FirewallCleanerSetter).SetFirewallCleaner(iptabler.NewCleaner(ctx, config))
```

Comment:
nit: The type match is a bit awkward 🙈 
We could avoid it with the "accept interfaces, return struct" idiom. 

That would require exporting `nftabler` struct though, which wouldn't be that big of a deal I think?

Alternatively, perhaps the `NewNftabler` should just accept a cleaner directly, so we don't need to set it from outside?